Source/Engine/Editor/VQAssetBrowser.h

```diff
@@ -0,0 +1,139 @@
+//	VQE
+//	Copyright(C) 2020  - Volkan Ilbeyli, Anilcan Gulkaya
+//
+//	This program is free software : you can redistribute it and / or modify
+//	it under the terms of the GNU General Public License as published by
+//	the Free Software Foundation, either version 3 of the License, or
+//	(at your option) any later version.
+//
+//	This program is distributed in the hope that it will be useful,
+//	but WITHOUT ANY WARRANTY; without even the implied warranty of
+//	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+//	GNU General Public License for more details.
+//
+//	You should have received a copy of the GNU General Public License
+//	along with this program.If not, see <http://www.gnu.org/licenses/>.
+//
+//	Contact: volkanilbeyli@gmail.com, anilcangulkaya7@gmail.com
+
+#pragma once
+#include "VQEditor.h"
+#include "../../Renderer/Renderer.h"
+
+namespace StringHash
+{
+	// very fast hash function + compile time
+	// http://www.cse.yorku.ca/~oz/hash.html
+	inline constexpr uint32 Basic(const char* str)
```

Comment:
image in my engine:
![image](https://user-images.githubusercontent.com/48527900/170693160-8a85e45c-e8bb-4e31-a496-d4e3637de9d3.png)
I'm not using string comparison because it could be hard to read.

Reply:
I don't understand why string comparison would be hard to read.

```
if (extension == ".png") doPNGStuff();
if (extension == ".jpg") doJPGStuff();
...
```
^ I dont see any readability issues with this -- it just has some unnecessary string comparison steps.

This is why I think unordered_map is the better option here

```
return mExtentionToIconLookup.at(extension);
```

Pretty straightforward.

This also makes the function not really necessary -- just lookup the map and be done with it.